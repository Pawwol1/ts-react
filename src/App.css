* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

:root {
    --bg-color: linear-gradient(45deg, lightblue, lightyellow);
    --bg-box-color: linear-gradient(90deg, lightblue, lightyellow);
    --text-color: rgba(0, 0, 0, 0.7);
    --boxshadow: 8px 8px 24px 0px rgba(66, 68, 90, 1);
}