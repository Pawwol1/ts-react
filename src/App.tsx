import React from 'react';
import Header from './components/header/header';
import PageNotFound from './components/pageNotFound/pageNotFound';
import Sidebar from './components/sidebar/sidebar';
import UserDetails from './components/userDetails/userDetails';
import Userlist from './components/userlist/userlist';
import UserlistSearch from './components/userlistSearch/userlistSearch';


function App() {
  return (
    <>
      <Header/>
      <Userlist/>
      <UserlistSearch/>
      <UserDetails/>
      <PageNotFound/>
      <Sidebar/>
    </>
  );
}

export default App;
