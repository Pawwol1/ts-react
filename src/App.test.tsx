import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'
import App from './App';

it('snapshot for App component', () => {
  const renderedComponent = renderer.create(
      <MemoryRouter initialEntries={['/']}>
          <App />
      </MemoryRouter>).toJSON()
  expect(renderedComponent).toMatchSnapshot()
})

test('should show "Page not found"', () => {
  const badRoute = '/bad/route'
  render(
      <MemoryRouter initialEntries={[badRoute]}>
          <App />
      </MemoryRouter>)
  expect(screen.getByText(/Page not found/i)).toBeInTheDocument()
})