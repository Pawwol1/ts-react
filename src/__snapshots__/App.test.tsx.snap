// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for App component 1`] = `
Array [
  <header>
    <div
      className="header_box"
    >
      <h1
        className="header_box--title"
      >
        Meet ReqRes users
      </h1>
      <p
        className="header_box--text"
      >
        Application created with free ReqRes API
      </p>
    </div>
  </header>,
  <h3
    className="sidebar_title"
  >
    Please choose which side you want to see the surprise on:
  </h3>,
  <div
    className="sidebar_open"
  >
    <div>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        left
      </button>
    </div>
    <div>
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        right
      </button>
    </div>
  </div>,
  <div
    className="userlist_box"
  >
    <div
      className="userlistSearch_box"
    >
      <div
        className="userlistSearch_box--title"
      >
        <h2
          className="userlistSearch_box--title--text"
        >
          Find a user
        </h2>
      </div>
      <div
        className="userlistSearch_box--inputwrapper"
      >
        <input
          onChange={[Function]}
          placeholder="Type to search"
          type="text"
          value=""
        />
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root Mui-disabled  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          Search
          <span
            className="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27c1.2-1.4 1.82-3.31 1.48-5.34-.47-2.78-2.79-5-5.59-5.34-4.23-.52-7.79 3.04-7.27 7.27.34 2.8 2.56 5.12 5.34 5.59 2.03.34 3.94-.28 5.34-1.48l.27.28v.79l4.25 4.25c.41.41 1.08.41 1.49 0 .41-.41.41-1.08 0-1.49L15.5 14zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="userlistSearch_box--userlist"
      >
        <div
          className="userlistSearch_box--userlist--notfound"
        >
          No user found
        </div>
      </div>
    </div>
    <div>
      <p
        aria-hidden="true"
        data-testid="hidden"
      />
    </div>
  </div>,
]
`;
